src/Magento/FunctionalTestingFramework/Util/TestGenerator.php

```diff
@@ -319,45 +342,67 @@ private function generateInjectMethod()
+        $this->notGeneratedTestsArray = [];
+
         /** @var TestObject[] $testObjects */
         $testObjects = $this->loadAllTestObjects($testsToIgnore);
-        $cestPhpArray = [];
+
         $filters = MftfApplicationConfig::getConfig()->getFilterList()->getFilters();
         /** @var FilterInterface $filter */
         foreach ($filters as $filter) {
             $filter->filter($testObjects);
         }
 
+        $removeLastTest = false;
         foreach ($testObjects as $test) {
-            // Do not generate test if it is an extended test and parent does not exist
-            if ($test->isSkipped() && !empty($test->getParentName())) {
-                try {
-                    TestObjectHandler::getInstance()->getObject($test->getParentName());
-                } catch (TestReferenceException $e) {
-                    print("{$test->getName()} will not be generated. Parent {$e->getMessage()} \n");
-                    continue;
+            try {
+                // Do not generate test if it is an extended test and parent does not exist
+                if ($test->isSkipped() && !empty($test->getParentName())) {
+                    try {
+                        TestObjectHandler::getInstance()->getObject($test->getParentName());
+                    } catch (TestReferenceException $e) {
+                        print("{$test->getName()} will not be generated. Parent {$e->getMessage()} \n");
+                        continue;
+                    }
                 }
-            }
 
-            $this->debug("<comment>Start creating test: " . $test->getCodeceptionName() . "</comment>");
-            $php = $this->assembleTestPhp($test);
-            $cestPhpArray[] = [$test->getCodeceptionName(), $php];
+                $this->debug("<comment>Start creating test: " . $test->getCodeceptionName() . "</comment>");
+                $php = $this->assembleTestPhp($test);
+                $this->cestPhpArray[] = [$test->getCodeceptionName(), $php];
+                $removeLastTest = true;
 
-            $debugInformation = $test->getDebugInformation();
-            $this->debug($debugInformation);
-            $this->debug("<comment>Finish creating test: " . $test->getCodeceptionName() . "</comment>" . PHP_EOL);
+                $debugInformation = $test->getDebugInformation();
+                $this->debug($debugInformation);
+                $this->debug("<comment>Finish creating test: " . $test->getCodeceptionName() . "</comment>" . PHP_EOL);
 
-            //write to manifest here if manifest is not null
-            if ($testManifest != null) {
-                $testManifest->addTest($test);
+                //write to manifest here if manifest is not null
+                if ($testManifest != null) {
+                    $testManifest->addTest($test);
+                }
+            } catch (\Exception $e) {
+                $this->notGeneratedTestsArray[] = [$test->getName() => $e->getMessage()];
```

Comment:
can number of tests that failed to generate due to schema validation error and annotation errors be printed in a similar format in mftf log?

Reply:
It's print in similar way with `number + word`. Let me know if you mean something else.

Reply:
I introduced a schema error in `AdminLoginFailedTest` 

I was expecting to also see something like the below:
1. Error: Failed to generate {$test->getName()}
2. All tests generated except for AdminLoginFailedTest


What I see:
1. FAILURE: Schema validation error Element 'actionGroup1': This element is not expected.Line: 23
2. No tests generated

Reply:
I introduced an annotation error by copying annotation block from `AdminLoginSuccessfulTest` to `AdminLoginFailedTest` (duplicate error)
I was expecting to see something like the below:
1. 2 tests failed to generate.
2. All tests generated except for `AdminLoginFailedTest` and `AdminLoginSuccessfulTest`

What I see:
1. ERROR: 1 Suite failed to generate. See mftf.log for details.
2. No tests generated

Reply:
What I see:
All tests generate when there is an annotation error
No tests generate when there is a schema validation error

I was expecting:

- All tests except the test xmls with annotation error generate (will need to evaluate if test runs with incorrect annotations can cause any run time failures)
- All tests except the test xmls with schema validation error generate (if this can't be achieved, maybe a hard failure is still the better solution)

Reply:
see reply in general comment